hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestStreamWriteOperatorCoordinator.java

```diff
@@ -233,48 +233,49 @@ void testSyncMetadataTable() throws Exception {
     assertThat(completedTimeline.lastInstant().get().getTimestamp(), startsWith(HoodieTableMetadata.SOLO_COMMIT_TIMESTAMP));
 
     // test metadata table compaction
-    // write another 4 commits
-    for (int i = 1; i < 5; i++) {
+    // write another 9 commits to trigger compaction twice. Since default clean version to retain is 2.
```

Comment:
@danny0405 : can you review changes in flink classes.

Reply:
would rather prefer to set the clean version to retain as 1. Doing 9 commits is going to add to test time which is already quite high.

Reply:
Why we need to change the clean strategy then?

Reply:
Clean's retainFileVersions value is changed to 2 and it is actually hardcoded in HoodieMetadataWriteUtils and both Spark and Flink gets the writeConfig using this. So, this cannot be overridden.
The reason for making the change is to support restore of main table timeline for atleast 10 commits. Restore on main table also causes restore on metadata table. If the retainFileVersions is less than 2, then next cleaner job after compaction would remove previous file slice there by blocking restore on metadata table or loosing data.

Reply:
> The reason for making the change is to support restore

First of all, I'm confused why this change is related with restore ? The change is for MDT log compaction right? Can we address the restore issue in another PR ?

> then next cleaner job after compaction would remove previous file slice there by blocking restore on metadata table or loosing data.

Does the new cleaning strategy solve the issue, even we keep at least   2 versions for each file group, it does not ensure we can restore to a long history commit.

Reply:
Discussed offline, reverting this change.